client/my-sites/hosting/staging-site-card/index.js

```diff
@@ -227,14 +252,24 @@ const StagingSiteCard = ( { currentUserId, disabled, siteId, siteOwnerId, transl
 	);
 };
 
-export default connect( ( state ) => {
-	const currentUserId = getCurrentUserId( state );
-	const siteId = getSelectedSiteId( state );
-	const siteOwnerId = getSelectedSite( state )?.site_owner;
+export default withMediaStorage(
+	connect( ( state, { mediaStorage } ) => {
+		const currentUserId = getCurrentUserId( state );
+		const siteId = getSelectedSiteId( state );
+		const siteOwnerId = getSelectedSite( state )?.site_owner;
 
-	return {
-		currentUserId,
-		siteId,
-		siteOwnerId,
-	};
-} )( localize( StagingSiteCard ) );
+		let spaceQuotaExceededForStaging = false;
+		// We check against -1 as this is the default value for sites with
+		// upload_space_check_disabled option.
+		if ( mediaStorage.storage_used_bytes > -1 && mediaStorage?.max_storage_bytes > -1 ) {
+			spaceQuotaExceededForStaging =
+				mediaStorage.storage_used_bytes > mediaStorage.max_storage_bytes / 2;
+		}
```

Comment:
Why do we sometimes add `?` to `mediaStorage ` and sometimes not?

The only use of `withMediaStorage` is here, and it considers it can be `null`/`undefined`. We probably need to use `?` in all the cases.

https://github.com/Automattic/wp-calypso/blob/31e64d93932ce9287e7e2e3702249c1e52fa3bc2/client/my-sites/site-settings/media-settings-performance.jsx#L142-L150

I would also prefer to use `useMediaStorageQuery` directly, as shown in this Readme. But it's just my preference.

https://github.com/Automattic/wp-calypso/blob/9a07fbfff52b55c5dd2a31a1b65a661cfde37f02/client/blocks/plan-storage/README.md#L28

Reply:
I added the optional chaining operator to `storage_used_bytes`. However, it's not needed inside the if statement as it already exists. Thanks for pointing this out!

Reply:
@wojtekn , @sejas , as things are working, can someone approve this? I want to merge it first thing tomorrow.

Reply:
Let's hold with this for now, and let's discuss https://github.com/Automattic/dotcom-forge/issues/1882#issuecomment-1477545181

Reply:
Based on the related discussion conclusion, let's land it 🚀